ddtrace/tracer/tracer_test.go

```diff
@@ -1192,3 +1192,73 @@ func BenchmarkTracerStackFrames(b *testing.B) {
 		span.Finish(StackFrames(64, 0))
 	}
 }
+
+type testTracerStatsd struct {
```

Comment:
We should have only one `testStatsdClient` implementation. Please improve the other one instead of adding new implementations for each test. A test statsd client should be able to give you all calls to it _and_  be concurrent safe. For example, for a `Count` implementation there should be a counterpart `CountCall(name string) []countCall` method which returns all the count calls for it, e.g.:
```
type countCall struct {
    name  string
    value int64
    tags  []string
    rate  float64
}
```
The added statsd code should be in `statsd.go` and the mock (and if needed tests) should be in `statsd_test.go`

Reply:
Here's an example. But I think it'd be nicer if we don't prefix methods with `Get*`: https://github.com/DataDog/datadog-agent/blob/gbbr/table-names/pkg/trace/test/testutil/statsd.go